homework-submission/src/App.js

```diff
@@ -0,0 +1,136 @@
+import React, { Component } from "react";
+import "./App.css";
+import todosJSON from "./Sources/todosJSON.json";
+import List from "./Components/List";
+import NewTodoForm from "./Components/NewTodoForm";
+import uuid from "uuid/v4";
+
+const TODO_TEMPLATE = {
+  description: "",
+  deadline: null,
+  id: null,
+  done: false
+};
+
+export default class App extends Component {
+  state = {
+    todos: [...todosJSON],
+    updatedTodo: { ...TODO_TEMPLATE },
+    newTodo: {
+      ...TODO_TEMPLATE
+    }
+  };
+
+  handleToggleCheck = item => {
+    const todos = [...this.state.todos];
+    const todosCopy = todos.map(todo => {
+      if (todo.id !== item.id) return todo;
+      return { ...todo, done: !todo.done };
+    });
+
+    this.setState({ todos: todosCopy });
+  };
+
+  handleNewDescription = e => {
+    const newTodo = { ...this.state.newTodo };
+    newTodo.description = e.target.value;
+
+    this.setState({ newTodo });
+  };
+
+  handleNewDeadline = e => {
+    // is using a ref better here ? then combining this and handeleNewDescription in one go
+    let newTodo = { ...this.state.newTodo };
+    newTodo.deadline = e.target.value;
+
+    this.setState({ newTodo });
+  };
+
+  handleSubmit = e => {
+    e.preventDefault();
+    let newTodo = { ...this.state.newTodo };
+    if (!newTodo.description || !newTodo.deadline) return;
+
+    let { todos } = this.state;
+    newTodo = { ...newTodo, id: newTodo.id || uuid() };
+    todos = [...todos, newTodo];
```

Comment:
you are saving a reference to `state.todos` and then mutating it here, I'm sure you just got caught up in refactoring :)
you can simply do `const todos = [...this.state.todos, newTodo]` without the intermediate [destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment)

Reply:
indeed, got caught up in refactoring :)
fixed it though, thanks Caludio